lib/SILOptimizer/Differentiation/PullbackCloner.cpp

```diff
@@ -1616,6 +1616,74 @@ class PullbackCloner::Implementation final
     builder.emitZeroIntoBuffer(uccai->getLoc(), adjDest, IsInitialization);
   }
 
+  void visitInjectEnumAddrInst(InjectEnumAddrInst *inject) {
+    SILBasicBlock *bb = inject->getParent();
+    SILValue origEnum = inject->getOperand();
+    bool unsupported = false;
+
+    // Only `Optional`-typed operands are supported for now. Diagnose all other
+    // enum operand types.
+    //
+    auto *optionalEnumDecl = getASTContext().getOptionalDecl();
+    if (origEnum->getType().getEnumOrBoundGenericEnum() != optionalEnumDecl) {
+      LLVM_DEBUG(getADDebugStream()
+                 << "Unsupported enum type in PullbackCloner: " << *inject);
+      unsupported = true;
+    }
+
+    InitEnumDataAddrInst *origData = nullptr;
+    for (auto use : origEnum->getUses()) {
+      if (auto *init = dyn_cast<InitEnumDataAddrInst>(use->getUser())) {
+        // We need a more complicated analysis when init_enum_data_addr and
+        // inject_enum_addr are in different blocks, or there is more than one
+        // such instruction. Bail out for now.
+        //
+        if (origData || init->getParent() != bb) {
+          LLVM_DEBUG(getADDebugStream()
+                     << "Could not find a matching init_enum_data_addr for: "
+                     << *inject);
+          unsupported = true;
+          break;
+        }
+
+        origData = init;
+      }
+    }
+
+    if (unsupported) {
```

Comment:
If would fold this into two cases above. Makes more readable code

Reply:
Done.